Donation-platform/src/main/java/com/google/servlets/PersonalizationServlet.java

```diff
@@ -51,26 +51,42 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     // If no tags were selected, set the tags list to an empty list.
     if(requestData.equals("[]")) {
-      tags = Arrays.asList();
+      tags = new ArrayList(Arrays.asList());
     // Otherwise, set the tags list to the processed list of tags.
     } else {
       // Splits requestData (i.e. ["hunger","education","children"]) by non-alphanumeric characters
       // (except for spaces since a tag such as "racial equality" should not be split into "racial" 
       // and "equality") using a regex, and strips out the leading delimiter before doing so in order
       // to prevent split() from creating a leading empty string.      
-      tags = Arrays.asList(requestData.replace("[\"", "").split("[^\\w ]+"));
+      tags = new ArrayList(Arrays.asList(requestData.replace("[\"", "").split("[^\\w ]+")));
+    }
+
+    // Fixes formatting of the tag "Arts, Culture, Humanities" if selected.
+    // (The split() above uses commas as a delimeter and thus separates this
+    // tag into three tags ("Arts", "Culture", "Humanities"). Using a different
+    // was considered; however, the comma is the only character separating tags).
+    if(tags.contains("Arts")) {
+      tags.remove(" Culture");
+      tags.remove(" Humanities");
```

Comment:
what is this hardcoding for?

Reply:
My comment in lines 64-67 explains my thinking here. I agree hardcoding isn't ideal, but I didn't see a way around this issue (and considering that tags are static, I didn't think it'd be problematic long-term). I'm definitely open for a cleaner way to address this issue though.